### ISSUE TITLE
Implementation of design mockups

### ISSUE THREAD

**Issue by gbozee** [pyung/pyconng#9]
Work has begun on the conversion of the mockups to actual html and css.
We would be making use of bootstrap 4 as well as sass for organizational purposes and also since bootstrap is very common.
The design conversion is located in the design subfolder in the root.
The mockups are hosted at invision which makes it possible to get all the dimensions and assets that would be required to convert to an html/web implementation. If you are yet to gain access to this, send me your email so that I can add you.
**If you are in the frontend team, work has started. **
Here are a list of the desgins
https://invis.io/QNF9VN03C#/272711625_Home_Page
https://invis.io/QNF9VN03C#/272717284_Styleguide
https://invis.io/QNF9VN03C#/272717375_Footer
https://invis.io/QNF9VN03C#/272717487_Nav-Bar
https://invis.io/QNF9VN03C#/272711618_Purchase_Ticket_-_Complete
https://invis.io/QNF9VN03C#/272711619_Purchase_Ticket_-_Step_1_Copy_2
https://invis.io/QNF9VN03C#/272711620_Purchase_Ticket_-_Step_1_Copy
https://invis.io/QNF9VN03C#/272711621_Purchase_Ticket_-_Step_1
https://invis.io/QNF9VN03C#/272711622_Code_Of_Conduct
https://invis.io/QNF9VN03C#/272711623_Schedule
https://invis.io/QNF9VN03C#/272711624_Speaker

**Comment by gbozee** [comment]
Both desktop and mobile designs would be handled simultaneously So please take note.

Also, I have begun work on the home page, the navigation bar and some parts of the jumbotron has been implemented.

**Comment by gbozee** [comment]
If you are not familiar with how invision works please let me know.

**Comment by danidee10** [comment]
I've started work on the speakers page. I'm just dropping this here so someone else doesn't waste time and effort working on it.

Everyone should follow suit. If you want to work on a particular page/section make sure you drop a comment so everyone's aware.

**Comment by gbozee** [comment]
that's good @danidee. Also to avoid dedication of work, functionality similar across pages should be reused, I.e CSS sections like nav

**Comment by gbozee** [comment]
I have also fixed the issue with the design branch being in `master` instead of develop?

**Comment by depecode** [comment]
I am working on the code of conduct page......

**Comment by danidee10** [comment]
I'm working on the footer.

**Comment by gbozee** [comment]
I just pushed some changes to the develop branch.

The counter at the right. this is visible on the home page so please propagate to respective pages.

Also, I kinda began the process to split the style files by section/pages

**Comment by danidee10** [comment]
I thought those line numbers were part of Invision...lol :rofl:

**Comment by danidee10** [comment]
I'm working on the Purchase Ticket (Step 1 and Step 2) pages

**Comment by gbozee** [comment]
@danidee you are moving way faster than the rest of us 😁 great job

**Comment by gbozee** [comment]
I just added the ability to include and extend html snippets like how django or jinja would support this when we actually move the designs to make use of django templates
2771e20dd56b461079a6d2f5e88e2a16fb174228

the syntax for `include` looks like this

```
<include src="./navigation.html">
```
as for the `extend` there is a [base.html](https://github.com/pyung/pyconng/blob/develop/design/base.html) and an example of how to extend this template is available like this

```
<extends src="base.html">
<block name="title">Hello world </block>
</extends>
``` 
I have transformed both the html and the speakers.html template so @danidee10 for the other pages you have worked on, just adjust accordingly.

**Comment by yemisi-o** [comment]
I'm working on the Schedule page

**Comment by danidee10** [comment]
I'm working on the Purchase ticket step 1 page (The main page that shows the pricing)

**Comment by danidee10** [comment]
@gbozee i've not been able to get the new setup to work. I've installed the new dependencies but they're partially working. The includes work but `parcel` doesn't transform the CSS and some images. I've attached a screenshot
![parcel pycon](https://user-images.githubusercontent.com/11032373/35489296-3610322c-0495-11e8-9bbc-0147ad0b80c1.png)

**Comment by gbozee** [comment]
U would need to update ur parcel installation. `npm install -g parcel-bundler`

**Comment by gbozee** [comment]
@yemisi cool. If there are any issues with setup please let me know

**Comment by depecode** [comment]
@gbozee I have update the` npm install -g parcel-bundler` ... It give this error  `(node:6188) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): SyntaxError: 'super' keyword unexpected here`

**Comment by gbozee** [comment]
@depecode  are you still running into the issue above?

**Comment by danidee10** [comment]
@gbozee it's working now :+1:

**Comment by depecode** [comment]
@gbozee No... But this is the issues am facing now..... `Cannot resolve dependency './js\w3.js'

**Comment by danidee10** [comment]
@depecode you can remove the dependency on `w3.js` from `base.html` it's not needed.

**Comment by depecode** [comment]
Thanks @gbozee  and @danidee10 it is resolved.....

**Comment by depecode** [comment]
I am working on Purchase Ticket - Complete page....

**Comment by danidee10** [comment]
I'm working on the Main Ticket page (That lists all the categories and their prices)

**Comment by danidee10** [comment]
I'm Done with the ticket page. We're almost done it's remaining the Code of conduct. @depecode are you still working on the Code of conduct page?

**Comment by depecode** [comment]
@danidee10 I am done with the code of conduct.... I am working on Purchase Ticket - Complete page

**Comment by danidee10** [comment]
Oh! that's true, I Forgot that I commented on your PR. : man_facepalming: Great job! :+1:

**Comment by gbozee** [comment]
for mobile designs, something to take note of. the number section at the right doesn't show.

**Comment by danidee10** [comment]
I think we should keep it on mobile devices, It gives the page a little swag :smile:  and it doesn't interfere with the main content. 
It's not present on some pages already e.g The Purchase tickets page and the Purchase tickets complete page.

**Comment by depecode** [comment]
@gbozee  Please I need a calender icon on the purchase tickets complete or can I use fontawesome icon...

**Comment by gbozee** [comment]
you can use font-awesome for now. I would try and get the icons.

**Comment by gbozee** [comment]
Just completed the designs for the home page. A page not existing in the design but is very paramount is the internal pages.  I would be creating a new issue for us to track the implementation of setting up the site for call for proposals. So the following pages are important and need to be implemented as soon as possible.

1. Code of conduct page
2. The home page. (would remove sections related to pricing and speaker profile for now.)

**Comment by gbozee** [comment]
@danidee10 your implementation of the footer is kinda off. The footer is supposed to be broken into two sections so that different pages can use the lower part and not both?

**Comment by danidee10** [comment]
@gbozee Oh! I'll look at it tomorrow
